build.gradle

```diff
@@ -63,24 +62,25 @@ task extractRelease(type: Copy) {
 /** Remove native libs for platforms we do not support */
 task removeUnsupported(type: Delete) {
   dependsOn extractRelease
-  delete "${jarDir}/${pkgDir}/aix-ppc"
-  delete "${jarDir}/${pkgDir}/aix-ppc64"
-  delete "${jarDir}/${pkgDir}/freebsd-x86"
-  delete "${jarDir}/${pkgDir}/freebsd-x86-64"
-  delete "${jarDir}/${pkgDir}/linux-arm"
-  delete "${jarDir}/${pkgDir}/linux-armel"
-  delete "${jarDir}/${pkgDir}/linux-ppc"
-  delete "${jarDir}/${pkgDir}/linux-ppc64le"
-  delete "${jarDir}/${pkgDir}/linux-s390x"
-  delete "${jarDir}/${pkgDir}/linux-x86"
-  delete "${jarDir}/${pkgDir}/linux-mips64el"
-  delete "${jarDir}/${pkgDir}/openbsd-x86"
-  delete "${jarDir}/${pkgDir}/openbsd-x86-64"
-  delete "${jarDir}/${pkgDir}/sunos-sparc"
-  delete "${jarDir}/${pkgDir}/sunos-sparcv9"
-  delete "${jarDir}/${pkgDir}/sunos-x86"
-  delete "${jarDir}/${pkgDir}/sunos-x86-64"
-  delete "${jarDir}/${pkgDir}/win32-x86"
+  List<String> inclusions = ['darwin', 'linux-aarch64', 'linux-x86-64', 'win32-x86-64']
+  for (File arch : file("${jarDir}/${pkgDir}/").listFiles()) {
+    // skip non-directories
+    if (arch.isDirectory() == false) {
+      continue
+    }
+    // skip directories that do not contain native libraries
+    if (new File(arch, 'jnidispatch.dll').exists() == false &&
```

Comment:
Maybe we can do this in extractRelease with includes, instead of a separate task to remove the files we want?

Reply:
I think that's trickier because there's more than just those native libraries that we want to include (all of the directories that contain the class files too)? That's why there's logic around which files to delete.